hpcbench/driver.py

```diff
@@ -1084,6 +1086,28 @@ def popen(self, stdout, stderr):
         return subprocess.Popen([self._executor_script],
                                 stdout=stdout, stderr=stderr)
 
+    @contextlib.contextmanager
+    def module_env(self):
+        """Set current process environment according
+        to execution `environment` and `modules`
+        """
+        env = copy.copy(os.environ)
+        try:
+            for mod in self.execution.get('modules') or []:
+                module('load', mod)
```

Comment:
it's really cool that we have now this module API but I think it's a pity that it doesn't express a little more (cf. below)